beacon_node/lighthouse_network/src/peer_manager/peerdb.rs

```diff
@@ -291,6 +319,23 @@ impl<E: EthSpec> PeerDB<E> {
             .map(|(peer_id, _)| peer_id)
     }
 
+    /// Returns an iterator of all peers that are supposed to be custodying
+    /// the given subnet id that also belong to `allowed_peers`.
+    pub fn good_range_sync_custody_subnet_peer<'a>(
+        &'a self,
+        subnet: DataColumnSubnetId,
+        allowed_peers: &'a HashSet<PeerId>,
+    ) -> impl Iterator<Item = &'a PeerId> {
+        self.peers
+            .iter()
+            .filter(move |(peer_id, info)| {
```

Comment:
we no longer use the return value here, maybe we could optimise this function with `any` instead of `filter`, and return `bool` instead?

Reply:
This function is only used here to determine if there's any peers on a particular custody subnet, we don't really use the return value other than checking the count > 0 - so perhaps we could change this function to `has_good_range_sync_peer_on_custody_subnet` and short-circuit it (using `any`) without iterating through all peers (128 times for superdnoes)

https://github.com/sigp/lighthouse/blob/3182646d425dc497c024b636c8fdc870c9c65d2f/beacon_node/network/src/sync/range_sync/chain.rs#L1098-L1110

Reply:
(Also this is likely a micro optimisation I wouldn’t be too worried about for this PR)